app/models/image.rb

```diff
@@ -0,0 +1,14 @@
+class Image < ApplicationRecord
+  belongs_to :imageable, polymorphic: true, touch: true
+  attr_accessor :file
+
+  before_validation :populate_fields_from_file
+
```

Comment:
I think there should be some validation check here especially for the image size since all the other validation checks (which were written in the respective models for event, sponsor, and testimony) have been removed.

Reply:
Done